src/core_modules/capture-core/components/WidgetProfile/WidgetProfile.component.js

```diff
@@ -1,56 +1,90 @@
 // @flow
-import React, { type ComponentType, useState, useCallback } from 'react';
+import React, { type ComponentType, useState, useMemo } from 'react';
 import i18n from '@dhis2/d2-i18n';
 import { FlatList } from 'capture-ui';
-import cx from 'classnames';
 import { withStyles } from '@material-ui/core';
-import { colors } from '@dhis2/ui';
+import { useSelector } from 'react-redux';
+import { useDataQuery } from '@dhis2/app-runtime';
 import { Widget } from '../Widget';
 import type { Props } from './widgetProfile.types';
-import { useProfileInfo } from './hooks';
+
 
 const styles = {
-    flatListWrapper: {
-        padding: '0 16px',
-    },
-    itemRow: {
-        borderBottom: `1px solid${colors.grey400}`,
-        display: 'flex',
-        padding: '16px 0',
-        '&.isLastItem': {
-            borderBottomWidth: 0,
-        },
-    },
-    itemKey: {
-        width: 150,
-        color: colors.grey600,
+    profileWidgetWrapper: {
+        paddingBottom: '12px',
     },
+
 };
 const ProfileWidgetPlain = ({ classes }: Props) => {
-    const { attributes } = useProfileInfo();
+    const { teiId, programId } =
+      useSelector(({ router: { location: { query } } }) => ({ teiId: query.teiId, programId: query.programId }));
```

Comment:
Hey @huocha , 

I would advise to move the `useSelector` logic in the parrent of `ProfileWidgetPlain`.  You can pass` teiId and programId` as props to `ProfileWidgetPlain`. This way we ensure there is no redux logic in the WidgetProfile. 
What do you think? Thanks! 😃

Reply:
@simonadomnisoru I agree, I was thinking about moving that but forgot. just changed that :)